### DISCUSSION TITLE
Where are the strings added manually, and how do they work

### DISCUSSION THREAD

**Discussion by burner1024** [WeblateOrg/weblate#7511] [Q&A] [answered]
Apparently, when adding a variant to a unit, it's also added to the source language.
However, deleting the variant from translation does not then remove it from source language (still shown in UI).
Downloading and checking source language POT, the variant is not there. (Also checked repo status, it's clean).

1. Where is it?
2. Is it supposed to work like this?
3. What happens when POT is updated upstream and Weblate pulls it? Will all units added manually get discarded along with their translations? Or just source language ones? Or something else?

**Comment by nijel** [comment, marked as answer]
POT is typically not used for source strings. In bilingual formats, the source string is anything that exists at least in one language.

**Reply by burner1024** [reply]
I see, so, what's actually displayed in UI as source language is a virtual collection of all source strings existing in all translations in the component. And POT will never ever be changed by Weblate itself, correct?
UI is a bit misleading, because it still shows POT as translation file in component info and string info.
![Captura de pantalla de 2022-04-19 20-21-13](https://user-images.githubusercontent.com/1279852/164013572-5a893cc2-dbe9-4837-bcec-e8b6ffcd7d2d.png)
![Captura de pantalla de 2022-04-19 20-21-46](https://user-images.githubusercontent.com/1279852/164013579-24c0ba9c-245d-4876-bb5d-c162df5ea180.png)

But why then the variant is not removed from source after being deleted from actual language translation? Or is it something that happens later in background?

**Reply by nijel** [reply]
Yes, POT should not be changed by Weblate.

**Reply by burner1024** [reply]
Great. Sorry be to insistent, but I'd still like to clarify the other two questions:
1. Do you agree that the interface is misleading in this case? Should I open an issue about it?
2. Why the new string (a variant) is not removed from source language view after being deleted from actual language translation? Or is it something that happens later in background?

**Reply by nijel** [reply]
The string will be removed once the files are parsed again. It should be done immediately. Fixed in 48bcd02777c5a7eab387ac720fbfe4d058051d85
